Sorting/HeapSort.cpp

```diff
@@ -0,0 +1,71 @@
+/*
+    Heap Sort :
+    Uses an array-implemented heap to sort a list of numbers.
+
+    Time complexity:
+    O(N * log(N)), where n is the number of elements to be sorted.
+
+    Space complexity:
+    O(1).
+*/
+
+#include <iostream>
+#include <vector>
+
+#include "SortingUtils.h"
+
+using namespace std;
+
+//creates ordered heap
+void heapify(vector<int> &heap, int parent, const int size, const int order, const bool showState){
+   int child = (parent*2)+1, temp;
+   while (child<size){//travels down children
+       if (showState) displayState(heap);
+       if ((order * heap[child]) < (order * heap[child+1]) and (child+1 < size)){
+           child++;//child refers to the larger/smaller of the two children
+        }

```

Comment:
Add a single tab before starting a single-line comment, and add a single space before the first character in the comment.

Eg.
```C++
child++;   // child refers to the larger of the two children
           // (or smaller when sorting in descending order)
```